test/test_ruby_parser.rb

```diff
@@ -1401,16 +1401,47 @@ def test_qwords_empty
     assert_parse rb, pt
   end
 
-  def test_qwords_line_breaks
-    skip "not yet"
+  def test_array_line_breaks
+    # It seems like arrays are roughly created when a certain element is created
+    # In ruby > 1.9 it seems like that is after the last element, so the array
+    # itself is assigned line 3 (since the last element is on line 3) and for
+    # ruby <= 1.9 it seems to get created after the first element, so the array
+    # itself is assigned line 2 (since the first element is on line 2).
+    # This seems to happen since arrays like this are created with a line in
+    # ruby_parser.yy like `result = s(:array, val[0])`. So, the array is not
+    # created by itself. The creation of the array itself is deferred until there
+    # is an element to create it with. That seems to mess up line numbers
+    # for the array. Luckily, the arary elements all seemt to get the correct
+    # line number.
+    start_line = self.class.to_s =~ /1[89]/ ? 2 : 3
```

Comment:
This is the line that accounts for parser version. As has been noted throughout..this is super hacky. I only used this to surface the logic that seems to happen to account for the parser differences.